File: app/src/main/java/com/example/oderfoodapp/ViewHolder/FoodViewHolder.java
package com.example.oderfoodapp.ViewHolder;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.oderfoodapp.Interface.ItemClickListener;
import com.example.oderfoodapp.R;

public class FoodViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
    public TextView txtFoodName,txtFoodPrice;
    public ImageView FoodImageView,fav_img,img_cart;
    private ItemClickListener itemClickListener;

    public FoodViewHolder(@NonNull View itemView) {

        super(itemView);
        txtFoodName = itemView.findViewById(R.id.food_name);
        txtFoodPrice = itemView.findViewById(R.id.food_price);
        FoodImageView = itemView.findViewById(R.id.food_im);
        fav_img = itemView.findViewById(R.id.img_fav);
        img_cart = itemView.findViewById(R.id.img_cart);
        itemView.setOnClickListener(this);
    }

    public void setItemClickListener(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onClick(View view) {
        itemClickListener.onClick(view,getAdapterPosition(),false   );

    }
}

File: app/src/main/java/com/example/oderfoodapp/view/view_holder/FoodViewHolder.kt
package com.example.oderfoodapp.view.view_holder

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.oderfoodapp.`interface`.ItemClickListener
import com.example.oderfoodapp.R

class FoodViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
    var txtFoodName: TextView
    var txtFoodPrice: TextView
    var FoodImageView: ImageView
    var fav_img: ImageView
    var img_cart: ImageView
    private var itemClickListener: ItemClickListener? = null

    init {
        txtFoodName = itemView.findViewById(R.id.food_name)
        txtFoodPrice = itemView.findViewById(R.id.food_price)
        FoodImageView = itemView.findViewById(R.id.food_im)
        fav_img = itemView.findViewById(R.id.img_fav)
        img_cart = itemView.findViewById(R.id.img_cart)
        itemView.setOnClickListener(this)
    }

    fun setItemClickListener(itemClickListener: ItemClickListener?) {
        this.itemClickListener = itemClickListener
    }

    override fun onClick(view: View) {
        itemClickListener!!.onClick(view, adapterPosition, false)
    }
}